'supabase start' frequently fails with 'service not healthy'

# Bug report

## Describe the bug

Running `supabase start` on Github builders frequently fails with 'service not healthy'.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a Github action workflow with `supabase/setup-cli@v1` and `version: latest`.
2. In the workflow start the supabase setup with `supabase start`.
3. Optionally, for good measure, do this with 2 other supabase configurations (using different ports).

## Expected behavior

The builder starts cleanly without errors.

## Github action log

```
Run supabase start
  supabase start
  supabase status
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
Pulling images... (1/13)
Pulling images... (1/13)
Pulling images... (2/13)
Pulling images... (3/13)
Pulling images... (4/13)
Pulling images... (5/13)
Pulling images... (6/13)
Pulling images... (7/13)
Pulling images... (8/13)
Pulling images... (9/13)
Pulling images... (10/13)
Pulling images... (11/13)
Pulling images... (12/13)
Starting database...
Restoring branches...
Setting up initial schema...
Applying migration 20230105212858_initial.sql...
Seeding data supabase/seed.sql...
Starting containers...
Error: service not healthy: [supabase_storage_supabase_test supabase_pg_meta_supabase_test supabase_studio_supabase_test]
Try rerunning the command with --debug to troubleshoot the error.
Error: Process completed with exit code 1.
```

Unfortunately, I wasn't able to get a better log with `--debug`. With the debug flag, the action didn't exhibit the problem.

## System information

Github builder, ubuntu-latest (22.04).
See https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

## Additional context

Older versions don't show the error.

This was most likely introduced by the "fix" to https://github.com/supabase/cli/issues/146
https://github.com/supabase/cli/pull/770

--- a/internal/db/start/start.go
+++ b/internal/db/start/start.go
@@ -77,7 +77,11 @@ func StartDatabase(ctx context.Context, fsys afero.Fs, w io.Writer, options ...f
 	if _, err := utils.DockerStart(ctx, config, hostConfig, utils.DbId); err != nil {
 		return err
 	}
-	return initDatabase(ctx, fsys, w, options...)
+	err := initDatabase(ctx, fsys, w, options...)
+	if err != nil {
+		utils.DockerRemove(utils.DbId)
+	}
+	return err
 }
 
 func initDatabase(ctx context.Context, fsys afero.Fs, w io.Writer, options ...func(*pgx.ConnConfig)) error {

--- a/internal/start/start.go
+++ b/internal/start/start.go
@@ -478,7 +478,7 @@ func waitForServiceReady(ctx context.Context, started []string) error {
 		started = unhealthy
 		return len(started) == 0
 	}
-	if !reset.RetryEverySecond(probe, 10*time.Second) {
+	if !reset.RetryEverySecond(probe, 20*time.Second) {
 		return fmt.Errorf("service not healthy: %v", started)
 	}
 	return nil

